addons/portal_rating/controllers/thread.py

```diff
@@ -0,0 +1,29 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import NotFound
+
+from odoo.addons.mail.controllers import thread
+from odoo import http
+from odoo.http import request
+from odoo.tools import html2plaintext
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_update_content(
+        self, message_id, body, attachment_ids, attachment_tokens=None, partner_ids=None, **kwargs
+    ):
+        if rating_value := kwargs.pop("rating_value", False):
+            if not request.env["mail.message"]._get_with_access(message_id, "create", **kwargs):
```

Comment:
Why not removing the check (that is anyway not aligned with the base one in mail that checks people can edit message), and do everything after super, when we know the message is valid and people can edit it ? It is better to delegate the check once to the base code and rely on it to avoid code duplication.